### ISSUE TITLE
How can I send an IBC transfer transaction to solomachine?

### ISSUE THREAD

**Issue by shenao78** [chainapsis/keplr-wallet#827]
I want to send an IBC transfer transaction, and the destination chain's light client type on the source chain is solomachine. When I add the corresponding IBC Channel, I receive the prompt "Client is not for xxx". After looking at the source code of keplr wallet, I found that the reason is that when clicking on save, it retrieves the corresponding client state by the channel ID, and then compares whether clientState.chainId is equal to selectedChainId. However, the solomachine client state does not have the chainId field, which causes the check to fail and prevents the transaction from being sent.

<img width="351" alt="image" src="https://github.com/chainapsis/keplr-wallet/assets/8003221/62ace95a-f694-48ac-a861-059e95a8f8fd">

**Comment by HeesungB** [comment]
@shenao78 Solomachine is not considered in Keplr now. Do you need this function to make something?

**Comment by shenao78** [comment]
> @shenao78 Solomachine is not considered in Keplr now. Do you need this function to make something?

Yes, we are working on IBC heterogeneous cross-chain, but currently we are unable to implement the corresponding light client on the Cosmos chain. The current solution involves using a solomachine to validate the data packets and then sending the data signed by the solomachine to the Cosmos chain.

**Comment by HeesungB** [comment]
@shenao78 hello.

Our team didn't even consider solomachine when we created Keplr.

I don't think we can implement it right now as it is taking up all the development resources of the team.

We'll put it in the backlog and work on it.
